kclvm/compiler_base/error/src/lib.rs

```diff
@@ -1,5 +1,488 @@
-pub mod diagnostic;
+//! This crate provides `DiagnosticHandler` supports diagnostic messages to terminal stderr.
```

Comment:
The usage and design of ErrorHandler should be focused on in lib.rs, and the fluent template can be briefly introduced.

Reply:
Moved `DiagnosticHandler` to 'diagnostic_builder.rs' and add comments for `DiagnosticHandler` in the 'diagnostic_builder.rs'.
Added some comments for Compiler-Base-Error to lib.rs.